core/arch/riscv/kernel/boot.c

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2015-2021, Linaro Limited
+ * Copyright 2022 NXP
+ */
+
+#include <assert.h>
+#include <compiler.h>
+#include <config.h>
+#include <console.h>
+#include <keep.h>
+#include <kernel/boot.h>
+#include <kernel/linker.h>
+#include <kernel/misc.h>
+#include <kernel/panic.h>
+#include <kernel/thread.h>
+#include <riscv.h>
+#include <stdio.h>
+#include <trace.h>
+#include <util.h>
+
+#include <platform_config.h>
+
+#define PADDR_INVALID               ULONG_MAX
+
+uint32_t sem_cpu_sync[CFG_TEE_CORE_NB_CORE];
+
+static void init_runtime(unsigned long pageable_part __unused)
+{
+}
+
+void init_sec_mon(unsigned long nsec_entry __maybe_unused)
+{
+	assert(nsec_entry == PADDR_INVALID);
+	/* Do nothing as we don't have a secure monitor */
+}
+
+static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
+{
+	/*
+	 * Mask asynchronous exceptions before switch to the thread vector
+	 * as the thread handler requires those to be masked while
+	 * executing with the temporary stack. The thread subsystem also
+	 * asserts that the foreign interrupts are blocked when using most of
+	 * its functions.
+	 */
+	thread_set_exceptions(THREAD_EXCP_ALL);
+
+	/*
+	 * Pager: init_runtime() calls thread_kernel_enable_vfp() so we must
```

Comment:
Do you expect that arch/riscv ever will need to implement a pager?
I find it highly unlikely, better to keep things simple and assume that it will never happen.